Source/AwsCommonRuntimeKit/crt/ShutdownCallbackCore.swift

```diff
@@ -0,0 +1,85 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import AwsCCommon
+import AwsCAuth
+
+public typealias ShutdownCallback = () -> Void
+
+/// Core classes have manual memory management.
+/// You have to balance the retain & release calls in all cases to avoid leaking memory.
+class ShutdownCallbackCore {
```

Comment:
why not use the same naming as `CRT***` than introducing a new concept?

Reply:
This is an internal class.